### PR TITLE
Add DataLoader2-related online doc

### PR DESCRIPTION
**Pull request by ejguan**
Per title
You can check the generated doc in https://ejguan.github.io/dataloader2.html

### CODE REVIEW

File: docs/source/dataloader2.rst [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+Each ``ReadingServices`` would take the ``DataPipe`` graph and modify it to achieve a few features like dynamic sharding, sharing random seeds and snapshoting for multi-/distributed processes.
+
+This also allows easier transition of data-preprocessing pipeline from research to production. After the ``DataPipe`` graph is created and validated with the ``ReadingServices``, a different ``ReadingService`` that configures and connects to the production service/infra such as ``AIStore`` can be provided to :class:`DataLoader2` as a drop-in replacement. The ``ReadingService`` could potentially search the graph, and find ``DataPipe`` operations that can be delegated to the production service/infra, then modify the graph correspondingly to achieve higher-performant execution.
+
+The followings are interfaces for custom ``ReadingService``.
+
+.. autoclass:: ReadingServiceInterface
+    :members:
+
+.. autoclass:: CheckpointableReadingServiceInterface
```
Comment (NivekT) [inline docs/source/dataloader2.rst:44]:
```suggestion

The checkpoint/snapshotting feature is a work in progress. Here is the preliminary interface (small changes are likely):

.. autoclass:: CheckpointableReadingServiceInterface
```

**Review by NivekT — APPROVED** [review]
LGTM! Thanks!

We will add a tutorial in a later PR.

File: docs/source/dataloader2.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    :members:
+
+A sequence of graph utilities are provided to help users to define their own ``ReadingService`` and modify the graph:
```
[the remark this answers is not in the document]
Comment (ejguan) [reply docs/source/dataloader2.rst:47]: I can do that!

File: docs/source/dataloader2.rst [hunk 3]
```diff
@@ -0,0 +1,90 @@
+New DataLoader
```
[the remark this answers is not in the document]
Comment (ejguan) [reply docs/source/dataloader2.rst:1]: LMAO, I was thinking the same. But, I can't find a better to summarize all `DataLoader2`, `ReadingService`, etc. and we still have a specific sector called `DataLoader2`. 
Any other suggestions? (DataLoader2 related? -> still weird) If not, I will change it to `DataLoader2`.

File: docs/source/dataloader2.rst (hunk 1 above)
Comment (ejguan) [reply docs/source/dataloader2.rst:44]: Make sense

File: docs/source/dataloader2.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (NivekT) [reply docs/source/dataloader2.rst]: I think just `DataLoader2` should be fine.

**Pull request closed without merging**